evennia/contrib/worldloader/command.py

```diff
@@ -0,0 +1,281 @@
+    their own unless they do something particularly advanced.
+
+    A MUXCommand command understands the following possible syntax:
+
+      name[ with several words][/switch[/switch..]] arg1[,arg2,...] [[=|,] arg[,..]]
+
+    The 'name[ with several words]' part is already dealt with by the
+    cmdhandler at this point, and stored in self.cmdname. The rest is stored
+    in self.args.
+
+    The MuxCommand parser breaks self.args into its constituents and stores them
+    in the following variables:
+      self.switches = optional list of /switches (without the /)
+      self.raw = This is the raw argument input, including switches
+      self.args = This is re-defined to be everything *except* the switches
+      self.lhs = Everything to the left of = (lhs:'left-hand side'). If
+                 no = is found, this is identical to self.args.
+      self.rhs: Everything to the right of = (rhs:'right-hand side').
+                If no '=' is found, this is None.
+      self.lhslist - self.lhs split into a list by comma
+      self.rhslist - list of self.rhs split into a list by comma
+      self.arglist = list of space-separated args (including '=' if it exists)
+
+      All args and list members are stripped of excess whitespace around the
+      strings, but case is preserved.
+      """
+
+    def func(self):
+        """
+        This is the hook function that actually does all the work. It is called
+        by the cmdhandler right after self.parser() finishes, and so has access
+        to all the variables defined therein.
+        """
+        # this can be removed in your child class, it's just
+        # printing the ingoing variables as a demo.
+        super(MuxCommand, self).func()
+
+
+#------------------------------------------------------------
+# import csv tables
+#------------------------------------------------------------
+class CmdImportCsv(MuxCommand):
+    """
+    Usage:
+      @importcsv
+      
+      <modelname> must be in world_settings.CSV_DATA.
+      If <modelname> is empty, it will import all csv files in world_settings.CSV_DATA.
+    """
+    key = "@importcsv"
+    locks = "perm(Builders)"
+    help_cateogory = "Builders"
+    arg_regex = r"\s.*?|$"
+
+    def func(self):
+        "Implement the command"
+    
+        caller = self.caller
+        
+        # count the number of files loaded
+        count = 0
+        
+        # get app_name
+        app_name = world_settings.WORLD_DATA_APP
+        
+        # get model_name, can specify the model name in args
+        # if no args is given, load all models in settings.WORLD_DATA
+        models = self.args
+        if models:
+            models = [arg.strip() for arg in models.split(',')]
+        else:
+            models = world_settings.WORLD_DATA
+
+        # import models one by one
+        for model_name in models:
+            # can only import models in world_settings.WORLD_DATA
+            if not model_name in world_settings.WORLD_DATA:
```

Comment:
There are rooms, exits, common objects and maybe other types, each CSV file must match a db model. So I limited the models.